packages/blade/src/components/AutoComplete/_decisions/decisions.md

```diff
@@ -0,0 +1,240 @@
+<ActionList>
+  {cities.map((city) => <ActionListItem title={city} />)}
+</ActionList>
+```
+
+Complex flows such as [Adding New Item to ActionList](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=47091%3A711978&mode=dev) should be handled with Controlled API.
+
+<details>
+<summary>Show Full Code</summary>
+
+```jsx
+const allCities = ['Mumbai', 'Pune', 'Hyderabad'];
+
+const ControlledAutoComplete = () => {
+  // Controlled Filtering
+  const [cities, setCities] = React.useState(allCities);
+  // Controlled Input Value
+  const [currentInputValue, setCurrentInputValue] = React.useState('');
+  // Controlled Selections
+  const [currentSelections, setCurrentSelections] = React.useState<string[]>([]);
+
+  return (
+    <Dropdown>
+      <AutoComplete
+        label="Select City"
+        // Controlled input value
+        // Sets the input value of input element in AutoComplete
+        value={currentInputValue}
+        // Triggers when input value changes
+        onChange={(inputValue) => {
+          if (inputValue) {
+            // filtering logic
+            setCities(allCities.filter((city) => city.includes(inputValue)));
+          } else {
+            setCities(allCities);
+          }
+          setCurrentInputValue(inputValue);
+        }}
+        // Selected Items in AutoComplete
+        selected={currentSelections}
+        onSelectChange={({ values }) => {
+          setCurrentSelections(values);
+        }}
+      />
+      <DropdownOverlay>
+        <ActionList>
+          {cities.map((city) => (
+            <ActionListItem
+              title={city}
+              value={city.toLowerCase()}
+            />
+          ))}
+        </ActionList>
+      </DropdownOverlay>
+    </Dropdown>
+  );
+};
+```
+
+</details>
+
+### In BottomSheet
+
+> **Warning**
+>
+> We will have to turn BottomSheetHeader (and maybe BaseHeader) to a slot to allow adding Input element there.
+
+**Open Question:** Should we also remove `leading` `trailing` etc props from Header? since same can be implemented inside the slot. This is what we follow for Footer.
+
+<img height="200" alt="image" src="https://github.com/razorpay/blade/assets/30949385/6867b09a-1ae2-4c4a-a977-e254b45b5282">
+
+```jsx
+<Dropdown selectionType="multiple">
+  <SelectInput label="Select Emails" />
+  <BottomSheet>
+    <BottomSheetHeader title="Do you want this report in an email?">
+      <AutoComplete label="Add receiver's email addresses" />
+    </BottomSheetHeader>
+    <BottomSheetBody>
+      <ActionList>
+        <ActionListItem title="Mumbai" value="mumbai" />
+        <ActionListItem title="Pune" value="pune" />
+        <ActionListItem title="Hyderabad" value="hyderabad" />
+      </ActionList>
+    </BottomSheetBody>
+  </BottomSheet>
+</Dropdown>
+```
+
+### With Tags Outside
+
+When possible selections are more than 11, it is advised to put tags outside of the Input.
```

Comment:
why the magic number 11?

Reply:
Took it from design guidelines. Let me ask Abhishek 😆

Reply:
??

Reply:
we changed it to 10. Its just roughly the number of tags after which scrolling horizontally to deselect becomes difficult. If there are more tags and height is constrained, you have to scroll too much to reach the last tag